postmarketOS/pmbootstrap
# Close #256: Implement strict package building mode

### Description

Contrary to abuild, pmbootstrap only installs makedepends, and
keeps the installed packages around - both hacks save lots of time.

However, they may introduce missing makedepends in the APKBUILDs,
that the authors of the APKBUILDs do not notice because it works
for them.

This PR adss a strict mode, which will always clean the chroots
before building a package, and also remove all installed dependencies
after the package was built. You can use the following syntax to
only zap once, but build many packages at once:
`pmbootstrap build --strict hello-world 0xffff heimdall`

It also builds dependencies properly without leaving makedepends
behind.

### How to test

* Build a few packages with and without strict mode
* Try to build packages in strict mode when the dependencies are not built yet and check the log if the depends get properly installed/uninstalled before.
* If you have enough time, do `pmbootstrap zap -p` and `pmbootstrap install` on this branch, just to make sure.
